
# HG changeset patch
# User Andreas Schneider <mail@cynapses.org>
# Date 1220969337 -7200
# Node ID 3798b0f8c2aae18920ec054f346b193cb202aaa8
# Parent a801e6e669ad3c40c27fec20310bc1e841764d75
Improve FFMPEG find module for new include dir structure.

Older versions will stil be found too.

--- a/owbuild/FindFFMPEG.cmake	Wed Sep 10 13:46:28 2008 +0200
+++ b/owbuild/FindFFMPEG.cmake	Tue Sep 09 16:08:57 2008 +0200
@@ -22,27 +22,96 @@
   # in the FIND_PATH() and FIND_LIBRARY() calls
   if (${CMAKE_MAJOR_VERSION} EQUAL 2 AND ${CMAKE_MINOR_VERSION} EQUAL 4)
     include(UsePkgConfig)
-    pkgconfig(libavcodec _FFMPEG_INCLUDEDIR _FFMPEG_LIBDIR _FFMPEG_LDFLAGS _FFMPEG_CFLAGS)
+    pkgconfig(libavcodec _AVCODEC_INCLUDEDIR _AVCODEC_LIBDIR _AVCODEC_LDFLAGS _AVCODEC_CFLAGS)
+    pkgconfig(libavformat _AVFORMAT_INCLUDEDIR _AVFORMAT_LIBDIR _AVFORMAT_LDFLAGS _AVFORMAT_CFLAGS)
+    pkgconfig(libavutil _AVUTIL_INCLUDEDIR _AVUTIL_LIBDIR _AVUTIL_LDFLAGS _AVUTIL_CFLAGS)
+    pkgconfig(libpostproc _POSTPROC_INCLUDEDIR _POSTPROC_LIBDIR _POSTPROC_LDFLAGS _POSTPROC_CFLAGS)
+    pkgconfig(libswscale _SWSCALE_INCLUDEDIR _SWSCALE_LIBDIR _SWSCALE_LDFLAGS _SWSCALE_CFLAGS)
   else (${CMAKE_MAJOR_VERSION} EQUAL 2 AND ${CMAKE_MINOR_VERSION} EQUAL 4)
     find_package(PkgConfig)
     if (PKG_CONFIG_FOUND)
-      pkg_check_modules(_FFMPEG libavcodec)
+      pkg_check_modules(_AVCODEC libavcodec)
+      pkg_check_modules(_AVFORMAT libavformat)
+      pkg_check_modules(_AVUTIL libavutil)
+      pkg_check_modules(_POSTPROC libpostproc)
+      pkg_check_modules(_SWSCALE libswscale)
     endif (PKG_CONFIG_FOUND)
   endif (${CMAKE_MAJOR_VERSION} EQUAL 2 AND ${CMAKE_MINOR_VERSION} EQUAL 4)
 
-  find_path(FFMPEG_INCLUDE_DIR
+  find_path(AVCODEC_INCLUDE_DIR
     NAMES
       avcodec.h
     PATHS
-      ${_FFMPEG_INCLUDEDIR}
+      ${_AVCODEC_INCLUDEDIR}
       /usr/include
       /usr/local/include
       /opt/local/include
       /sw/include
     PATH_SUFFIXES
+      libavcodec
       ffmpeg
   )
-  mark_as_advanced(FFMPEG_INCLUDE_DIR)
+  mark_as_advanced(AVCODEC_INCLUDE_DIR)
+
+  find_path(AVUTIL_INCLUDE_DIR
+    NAMES
+      avutil.h
+    PATHS
+      ${_AVUTIL_INCLUDEDIR}
+      /usr/include
+      /usr/local/include
+      /opt/local/include
+      /sw/include
+    PATH_SUFFIXES
+      libavutil
+      ffmpeg
+  )
+  mark_as_advanced(AVUTIL_INCLUDE_DIR)
+
+  find_path(AVFORMAT_INCLUDE_DIR
+    NAMES
+      avformat.h
+    PATHS
+      ${_AVFORMAT_INCLUDEDIR}
+      /usr/include
+      /usr/local/include
+      /opt/local/include
+      /sw/include
+    PATH_SUFFIXES
+      libavformat
+      ffmpeg
+  )
+  mark_as_advanced(AVFORMAT_INCLUDE_DIR)
+
+  find_path(POSTPROC_INCLUDE_DIR
+    NAMES
+      postprocess.h
+    PATHS
+      ${_POSTPROC_INCLUDEDIR}
+      /usr/include
+      /usr/local/include
+      /opt/local/include
+      /sw/include
+    PATH_SUFFIXES
+      libpostproc
+      ffmpeg
+  )
+  mark_as_advanced(POSTPROC_INCLUDE_DIR)
+
+  find_path(SWSCALE_INCLUDE_DIR
+    NAMES
+      swscale.h
+    PATHS
+      ${_SWCALE_INCLUDEDIR}
+      /usr/include
+      /usr/local/include
+      /opt/local/include
+      /sw/include
+    PATH_SUFFIXES
+      libswscale
+      ffmpeg
+  )
+  mark_as_advanced(SWSCALE_INCLUDE_DIR)
 
   find_library(AVCODEC_LIBRARY
     NAMES
@@ -55,6 +124,7 @@
       /sw/lib
   )
   mark_as_advanced(AVCODEC_LIBRARY)
+
   find_library(AVUTIL_LIBRARY
     NAMES
       avutil
@@ -66,6 +136,7 @@
       /sw/lib
   )
   mark_as_advanced(AVUTIL_LIBRARY)
+
   find_library(AVFORMAT_LIBRARY
     NAMES
       avformat
@@ -77,6 +148,7 @@
       /sw/lib
   )
   mark_as_advanced(AVFORMAT_LIBRARY)
+
   find_library(POSTPROC_LIBRARY
     NAMES
       postproc
@@ -88,6 +160,7 @@
       /sw/lib
   )
   mark_as_advanced(POSTPROC_LIBRARY)
+
   find_library(SWSCALE_LIBRARY
     NAMES
       swscale
@@ -117,7 +190,11 @@
   endif (SWSCALE_LIBRARY)
 
   set(FFMPEG_INCLUDE_DIRS
-    ${FFMPEG_INCLUDE_DIR}
+    ${AVCODEC_INCLUDE_DIR}
+    ${AVFORMAT_INCLUDE_DIR}
+    ${AVUTIL_INCLUDE_DIR}
+    ${POSTPROC_INCLUDE_DIR}
+    ${SWSCALE_INCLUDE_DIR}
   )
 
   if (AVCODEC_FOUND)

